test/test_dcrt.py

```diff
@@ -32,104 +32,234 @@ def test_dcrt_lasso():
     rng = np.random.RandomState(2024)
     estimated_coefs = rng.rand(10)
 
-    res = dcrt_zero(
+    results = dcrt_zero(
         X,
         y,
         estimated_coef=estimated_coefs,
         screening=False,
-        verbose=True,
         statistic="residual",
         random_state=2024,
     )
-    assert len(res[1]) == 10
-    assert len(res[2]) == 10
+    vi = dcrt_pvalue(
+        results[0], results[1], results[2], results[3], selection_only=False
+    )
+    assert len(vi[0]) <= 10
+    assert len(vi[1]) == 10
+    assert len(vi[2]) == 10
 
     # Checking with refit
     results_refit = dcrt_zero(
         X,
         y,
         refit=True,
         screening=False,
-        verbose=True,
         statistic="residual",
         random_state=2024,
     )
-    assert len(results_refit[1]) == 10
-    assert len(results_refit[2]) == 10
+    vi_refit = dcrt_pvalue(
+        results_refit[0],
+        results_refit[1],
+        results_refit[2],
+        results_refit[3],
+        selection_only=False,
+    )
+    assert len(vi_refit[0]) <= 10
+    assert len(vi_refit[1]) == 10
+    assert len(vi_refit[2]) == 10
 
-    # Checking with scaled statistics
-    results_scaled = dcrt_zero(
+    # Checking with use_cv
+    results_use_cv = dcrt_zero(
         X,
         y,
-        scaled_statistics=True,
+        use_cv=True,
         screening=False,
-        verbose=True,
         statistic="residual",
         random_state=2024,
     )
-    assert len(results_scaled[1]) == 10
-    assert len(results_scaled[2]) == 10
+    vi_use_cv = dcrt_pvalue(
+        results_use_cv[0],
+        results_use_cv[1],
+        results_use_cv[2],
+        results_use_cv[3],
+        selection_only=False,
+    )
+    assert len(vi_use_cv[0]) <= 10
+    assert len(vi_use_cv[1]) == 10
+    assert len(vi_use_cv[1]) == 10
 
-    # Checking without verbose (returns only selected indices)
-    results_no_verbose = dcrt_zero(
+    # Checking with a provided covariance matrix
+    cov = LedoitWolf().fit(X)
+
+    results_covariance = dcrt_zero(
         X,
         y,
-        scaled_statistics=True,
+        sigma_X=cov.covariance_,
         screening=False,
-        verbose=False,
         statistic="residual",
         random_state=2024,
     )
-    assert len(results_no_verbose) <= 10
+    vi_covariance = dcrt_pvalue(
+        results_covariance[0],
+        results_covariance[1],
+        results_covariance[2],
+        results_covariance[3],
+        selection_only=False,
+    )
+    assert len(vi_covariance[0]) <= 10
+    assert len(vi_covariance[1]) == 10
+    assert len(vi_covariance[1]) == 10
 
-    # Checking with use_cv
-    results_use_cv = dcrt_zero(
-        X,
-        y,
-        use_cv=True,
-        screening=False,
-        verbose=True,
-        statistic="residual",
-        random_state=2024,
+
+def test_dcrt_lasso_center():
+    """
+    This function tests the dcrt function using the Lasso learner
```

Comment:
```suggestion
    Test the dcrt function using the Lasso learner
```

Reply:
dcrt is rather meant for classification problems, right ?

Reply:
I don't know.